primo/data_parser/metric_data.py

```diff
@@ -282,7 +294,30 @@ def effective_weight(self):
 class SetOfMetrics:
     """Container for storing metrics and submetrics"""
 
-    def __init__(self):
+    def __init__(
+        self, supported_metrics: Optional[Dict[str, _SupportedContent]] = None
+    ):
+        if supported_metrics is not None:
+            for val in supported_metrics.values():
+                if not val.is_submetric:
+                    self.register_new_metric(
+                        name=val.name,
+                        full_name=val.full_name,
+                    )
+
+                else:
+                    self.register_new_submetric(
+                        name=val.name,
+                        parent_metric=getattr(self, val.parent_metric),
+                        full_name=val.full_name,
+                    )
+                metric = getattr(self, val.name)
+                metric._required_data = val.required_data
```

Comment:
I agree that it is not clean. The reason why it is setup in this manner is the following. `_required_data` attribute is being used to distinguish supported metrics and user-defined metrics. In the `compute_priotiy_scores` method, we check if the metric has `_required_data` attribute. If yes, then it is a supported metric, so we do data processing for that metric. If it is not, then it is user's responsibility to perform data processing.
This can be improved, but I suggest leaving a TODO for now and work on it in the future.

Reply:
alright. I ll hold off for now.